### ISSUE TITLE
ruff can not fix indent error and more whitespaces on function parameters (but black works well)

### ISSUE THREAD

**Issue by lgf4591** [astral-sh/ruff#9001]
# Code
```python
def main(   num   ):
            print(f"the num is: {num}")
```

# ruff config
```toml
[tool.black]
target-version = ["py37"]
line-length = 120
skip-string-normalization = true

[tool.ruff]
target-version = "py37"
line-length = 120
indent-width = 2
select = [
  "A",
  "ARG",
  "B",
  "C",
  "DTZ",
  "E",
  "EM",
  "F",
  "FBT",
  "I",
  "ICN",
  "ISC",
  "N",
  "PLC",
  "PLE",
  "PLR",
  "PLW",
  "Q",
  "RUF",
  "S",
  "T",
  "TID",
  "UP",
  "W",
  "YTT",
]
ignore = [
  # Allow non-abstract empty methods in abstract base classes
  "B027",
  # Allow boolean positional values in function calls, like `dict.get(... True)`
  "FBT003",
  # Ignore checks for possible passwords
  "S105",
  "S106",
  "S107",
  # Ignore complexity
  "C901",
  "PLR0911",
  "PLR0912",
  "PLR0913",
  "PLR0915",
  # T201 `print` found
  "T201",
]
unfixable = [
  # Don't touch unused imports
  "F401",
]
```

**Comment by charliermarsh** [comment]
Have you tried running `ruff format`, instead of `ruff check`?

**Comment by lgf4591** [comment]
yes, but it doesn't work

**Comment by MichaReiser** [comment]
I tried to reproduce the issue with the given configuration using `ruff format` but haven't been able to. 

```bash
~/astral/test
➜ ls
main.py		pyproject.toml

~/astral/test
➜ ../ruff/target/debug/ruff format
warning: The following rules may cause conflicts when used with the formatter: `ISC001`. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding them to the `ignore` configuration.
warning: Detected debug build without --no-cache.
1 file reformatted

~/astral/test
➜ bat main.py
───────┬─────────────────────────────────────────────────────────────────────────────────────────────
       │ File: main.py
───────┼─────────────────────────────────────────────────────────────────────────────────────────────
   1   │ def main(num):
   2   │   print(f"the num is: {num}")
───────┴─────────────────────────────────────────────────────────────────────────────────────────────
```

Can you list the content of your directory and share how you invoke ruff?
